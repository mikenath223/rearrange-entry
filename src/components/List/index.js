import { DragDropContext, Droppable, Draggable } from "react-beautiful-dnd";

const List = ({ listItems }) => {

  return (
    <DragDropContext onDragEnd={this.onDragEnd}>
      <Droppable droppableId="droppable">
        {(provided, snapshot) => (
          <div
            {...provided.droppableProps}
            ref={provided.innerRef}
            style={getListStyle(snapshot.isDraggingOver)}>
            {listItems.map((item, index) => (
              <Draggable key={item.id} draggableId={item.id} index={index}>
                {(provided, snapshot) => (
                  <div
                    ref={provided.innerRef}
                    {...provided.draggableProps}
                    {...provided.dragHandleProps}
                    style={getItemStyle(
                      snapshot.isDragging,
                      provided.draggableProps.style
                    )}
                  >
                    {item.content}
                  </div>
                )}
              </Draggable>
            ))}
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </DragDropContext>
  );
}

export default List