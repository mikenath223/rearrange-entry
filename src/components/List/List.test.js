import { render } from '@testing-library/react';
import List from './index';

describe('List', () => {
  it('should match snapshot', () => {
    const { container } = render(<List
      listItems={[]}
      handleSetListItems={jest.fn()}
    />);

    expect(container).toMatchSnapshot();
  });
});
