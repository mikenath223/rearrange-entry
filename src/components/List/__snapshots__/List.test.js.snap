// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match snapshot 1`] = `
<div>
  <div
    data-rbd-droppable-context-id="0"
    data-rbd-droppable-id="droppable"
    style="background: lightgrey; padding: 8px; width: 250px; min-height: 50px;"
  />
</div>
`;
